### PR TITLE
drivers/ins: vectornav support

### PR DESCRIPTION
**Pull request by oystub**
- Include the latest vectornav driver from upstream main branch
- EKF2 selector ignores instances with a specific device id
- Disable one internal imu on cube orange

### CODE REVIEW

**Comment by oystub** [comment]
Will make new pr, splitting up steps

- Insert driver code from upstream (we can use 1.14 version of driver instead of main)
- Modify driver code to work with 1.13 (just commenting out some lines, increasing stack size for serial ports, some device id constants)
- ignoring ekf instance by device id
- disable one internal imu

**Review by hermankolden — COMMENTED** [review]
No issues found after investigation of possible weaknesses. Just a typo and a type thing left. Haven't reviewed driver stuff.

NOTES
Line 844: Will not use next-best EKF different IMU, but just best EKF. Under there, the instance selection can simply fail without trying next alternative. However, invalid EKFs won't even be considered for switchover because of the second block of code, so this should be OK.
LIne 759: Can _selected_instance be set somewhere else? No, it's only assigned in SelectInstance, and initialized to INVALID_INSTANCE.

**Comment by oystub** [comment]
Superseded by
https://github.com/aviant-tech/PX4-Autopilot/pull/77

File: boards/cubepilot/cubeorange/init/rc.board_sensors [hunk 1]
```diff
@@ -6,7 +6,7 @@ board_adc start
 
 # SPI4
 ms5611 -s -b 4 start
-icm20602 -s -b 4 -R 12 start
+#icm20602 -s -b 4 -R 12 start # Disable one of the internal IMUs, cannot do it in params with multi-ekf
```
[the remark this answers is not in the document]
Comment (oystub) [reply boards/cubepilot/cubeorange/init/rc.board_sensors:9]: R sets the orientation of the sensor.

File: src/modules/ekf2/EKF2Selector.cpp [hunk 2]
```diff
@@ -117,6 +117,21 @@ void EKF2Selector::PrintInstanceChange(const uint8_t old_instance, uint8_t new_i
...
+	auto ignore_id = _param_ekf2_sel_ign_id.get();
+
+	if (ignore_id > 0) {
+		uint32_t ignore_id_unsiged = static_cast<uint32_t>(ignore_id);
+
+		if (_instance[ekf_instance].accel_device_id == ignore_id_unsiged
+		    || _instance[ekf_instance].gyro_device_id == ignore_id_unsiged
+		    || _instance[ekf_instance].baro_device_id == ignore_id_unsiged
+		    || _instance[ekf_instance].mag_device_id == ignore_id_unsiged) {
+			return false;
```
[the remark this answers is not in the document]
Comment (oystub) [reply src/modules/ekf2/EKF2Selector.cpp:130]: Yes, to be able to ignore any sensor

**Pull request closed without merging**